microprofile/WFLY-14277_MicroProfile_JWT_1_2_Upgrade.adoc

```diff
@@ -0,0 +1,98 @@
+= WFLY-14277 Upgrade to support  MicroProfile JWT 1.2
+:author:            Darran Lofthouse
+:email:             darran.lofthouse@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+The MicroProfile JWT 1.2 release is only a minor upgrade upgrading from MicroProfile 1.1 with small enhancements to APIs and supported configuration options.
+
+The full list of changes in this release can be found at:
+ * https://github.com/eclipse/microprofile-jwt-auth/releases/tag/1.2
+
+Without duplicating the list from the specification release notes, each of the updates to the specification are implemented by the SmallRye JWT project - for
+this reason it is anticipated this enhancement will be delivered as a set of component upgrades.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-14277[WFLY-14277]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-1596[EAP7-1596]
+* https://issues.redhat.com/browse/ELY-2068[ELY-2068]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+* [ ] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+The component upgrades can be applied exclusively to the https://github.com/wildfly/wildfly[WildFly] project.
+
+=== Other Interested Projects
+
+The implementation for MicroProfile JWT 1.2 is contained within the https://github.com/smallrye/smallrye-jwt[SmallRye JWT] project.
+
+The https://github.com/wildfly-security/wildfly-elytron[WildFly Elytron] project contains dependencies on both 
+MicroProfile and SmallRye JWT.  However this update should not require code changes to WildFly Elytron.
+
+== Requirements
+
+=== Hard Requirements
+
+The following Maven artifact dependencies in WildFly will be updated to switch to MicroProfile JWT 1.2:
+ * org.eclipse.microprofile.jwt:microprofile-jwt-auth-api
+ * org.eclipse.microprofile.jwt:microprofile-jwt-auth-tck
+ * io.smallrye:smallrye-jwt
+ 
+Where each of these projects utilises additional dependencies this RFE will review the required dependencies and update WildFly accordingly.
```

Comment:
@darranl I see the module `io.smallrye.jwt` now contains the following resources:

- smallrye-jwt-3.0.0-RC3.jar
- smallrye-jwt-cdi-extension-3.0.0-RC3.jar
- smallrye-jwt-common-3.0.0-RC3.jar
- smallrye-jwt-http-mechanism-3.0.0-RC3.jar

what is the reason behind this?

Reply:
The SmallRye JWT project has been split into some smaller Maven artefacts instead of one large artefact so now we need to add the ones that contain the classes which were split out.